File: src/org/cappuccino_project/ide/intellij/plugin/lang/ObjJFileType.java
package org.cappuccino_project.ide.intellij.plugin.lang;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class ObjJFileType extends LanguageFileType {
    public static final String FILE_EXTENSION = "j";
    public static final ObjJFileType INSTANCE = new ObjJFileType();

    private ObjJFileType() {
        super(ObjJLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Objective-J Script";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "An Objective-J script file for the Cappuccino Web Framework";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return FILE_EXTENSION;
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return ObjJIcons.DOCUMENT_ICON;
    }
}

File: src/org/cappuccino_project/ide/intellij/plugin/lang/ObjJFileType.kt
package org.cappuccino_project.ide.intellij.plugin.lang

import com.intellij.openapi.fileTypes.LanguageFileType

import javax.swing.*

class ObjJFileType private constructor() : LanguageFileType(ObjJLanguage.INSTANCE) {

    override fun getName(): String {
        return "Objective-J Script"
    }

    override fun getDescription(): String {
        return "An Objective-J script file for the Cappuccino Web Framework"
    }

    override fun getDefaultExtension(): String {
        return FILE_EXTENSION
    }

    override fun getIcon(): Icon? {
        return ObjJIcons.DOCUMENT_ICON
    }

    companion object {
        val FILE_EXTENSION = "j"
        val INSTANCE = ObjJFileType()
    }
}
